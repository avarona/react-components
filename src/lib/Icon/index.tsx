import * as React from "react";
import classnames from "classnames";
import styles from "./styles.module.scss";

type Props = {
  addClass?: string;
  img: any;
  alt?: string;
};

const Icon = ({ addClass, img, alt }: Props) => {
  const SVG = typeof img === "object" && img;

  return SVG ? (
    <SVG className={classnames(addClass, styles.icon)} />
  ) : (
    <img className={classnames(addClass, styles.icon)} src={img} alt={alt} />
  );
};

export default Icon;
