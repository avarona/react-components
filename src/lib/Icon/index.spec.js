import React from "react";
import { shallow } from "enzyme";

import Icon from "./";

describe("Icon Component", () => {
  it("renders", () => {
    const wrapper = shallow(<Icon>Click Me!</Icon>);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
