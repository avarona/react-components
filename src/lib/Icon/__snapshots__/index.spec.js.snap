// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component renders 1`] = `"<img class=\\"icon\\"/>"`;
